test/controllers/sections/guarantor/GuarantorVatControllerSpec.scala

```diff
@@ -0,0 +1,212 @@
+        contentAsString(result) mustEqual view(form, Transporter, NormalMode)(dataRequest(request), messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
+
+      val userAnswersSoFar = emptyUserAnswers
+        .set(GuarantorRequiredPage, true)
+        .set(GuarantorArrangerPage, Transporter)
+        .set(GuarantorVatPage, "answer")
+
+      val application = applicationBuilder(userAnswers = Some(userAnswersSoFar)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, guarantorVatRoute)
+
+        val view = application.injector.instanceOf[GuarantorVatView]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form.fill("answer"), Transporter, NormalMode)(dataRequest(request), messages(application)).toString
+      }
+    }
+
+    "must redirect to the next page when valid data is submitted" in {
+
+      val userAnswersSoFar = emptyUserAnswers
+        .set(GuarantorRequiredPage, true)
+        .set(GuarantorArrangerPage, Transporter)
+
+      MockUserAnswersService.set().returns(Future.successful(userAnswersSoFar))
+
+      val application =
+        applicationBuilder(userAnswers = Some(userAnswersSoFar))
+          .overrides(
+            bind[GuarantorNavigator].toInstance(new FakeGuarantorNavigator(onwardRoute)),
+            bind[UserAnswersService].toInstance(mockUserAnswersService)
+          )
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, guarantorVatRoute)
+            .withFormUrlEncodedBody(("value", "answer"))
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual onwardRoute.url
+      }
+    }
+
+    "must redirect to the next page when the NONGBVAT link" in {
+      val userAnswersSoFar = emptyUserAnswers
+        .set(GuarantorRequiredPage, true)
+        .set(GuarantorArrangerPage, Transporter)
+
+      MockUserAnswersService.set().returns(Future.successful(userAnswersSoFar))
+
+      val application =
+        applicationBuilder(userAnswers = Some(userAnswersSoFar))
+          .overrides(
+            bind[GuarantorNavigator].toInstance(new FakeGuarantorNavigator(onwardRoute)),
+            bind[UserAnswersService].toInstance(mockUserAnswersService)
+          )
+          .build()
+
+      running(application) {
+        val request = FakeRequest(GET, controllers.sections.guarantor.routes.GuarantorVatController.onNonGbVAT(testErn, testLrn).url)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual onwardRoute.url
+      }
+    }
+
+    "must redirect to the guarantor arranger controller for a GET if no guarantor arranger value is found" in {
```

Comment:
Is it valuable to have a test when a POST is performed and there is no guarantor arranger value in the user answers?

Reply:
For the edge case where a user has multiple tabs open and goes back to a tab and submits the page (but on another tab they had removed the guarantor earlier answers)

Reply:
Added new test on https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/200043256a2d0ed855a4ad8eec404e1d65df6c19

Reply:
Updated in https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/84/commits/6d0fd9e7bfd6a21b0e37acc5510ecc31c5ed5b0b